slick/src/main/scala/slick/basic/BasicBackend.scala

```diff
@@ -368,6 +370,12 @@ trait BasicBackend { self =>
 
     private[BasicBackend] var currentSession: Session = null
 
+    private[BasicBackend] def priority(highPriority: Boolean): Priority = {
+      if (currentSession != null) HighPriority
```

Comment:
Yes, all reads/writes to/from non-volatile variables in a context are protected by the volatile sync variable as long as you follow the correct pattern.

 From it's documentation:

    /** A volatile variable to enforce the happens-before relationship (see
      * [[https://docs.oracle.com/javase/specs/jls/se7/html/jls-17.html]] and
      * [[http://gee.cs.oswego.edu/dl/jmm/cookbook.html]]) when executing something in
      * a synchronous action context. It is read when entering the context and written when leaving
      * so that all writes to non-volatile variables within the context are visible to the next
      * synchronous execution. */
    @volatile private[BasicBackend] var sync = 0